python/kserve/kserve/storage/storage.py

```diff
@@ -525,7 +525,11 @@ def _download_local(uri, out_dir=None):
     def _download_from_uri(uri, out_dir=None):
         url = urlparse(uri)
         filename = os.path.basename(url.path)
-        mimetype, encoding = mimetypes.guess_type(url.path)
+        # Determine if the symbol '?' exists in the path
+        if  mimetypes.guess_type(url.path)[0] is None:
```

Comment:
Also check `url.query != ''` ?

Reply:
Yes, I overlooked this check and added it now

Reply:
Can you also add a test case here
https://github.com/kserve/kserve/blob/master/python/kserve/kserve/storage/test/test_storage.py

Reply:
Of course, there is indeed a need for a test case here